### PR TITLE
Document Java version should be exact 1.8

### LINKED ISSUE #915: Document Java version should be exact 1.8
https://github.com/apache/incubator-iceberg/blob/720b240a3e6a32b6f01173ebcc448c2c6e32e13a/build.gradle#L43-L44

https://github.com/apache/incubator-iceberg/blob/720b240a3e6a32b6f01173ebcc448c2c6e32e13a/build.gradle#L82-L83

When I build with Java 13 environment it fails. We should document it at least in `README.md`.

### PR DESCRIPTION
**Pull request by TisonKun**
This closes #915 .

cc @rdblue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ Community discussions happen primarily on the [dev mailing list][dev-list] or on
 
 ### Building
 
-Iceberg is built using Gradle 5.4.1.
+Iceberg is built using Gradle 5.4.1 with *exact* Java 1.8.
```
Comment (rdblue) [inline README.md:55]: Isn't it enough to state that Iceberg is built with 1.8? Why add "exact" in bold?

**Comment by rdblue** [comment]
Thanks, @TisonKun! We might want to change this to a table soon, since the Java 11 PR, #577, appears to be almost ready (waiting for tests to pass).

File: README.md (hunk 1 above)
Comment (TisonKun) [reply README.md:55]: Make sense for marketing and we're working for Java 11. Updated

**Comment by rdblue** [comment]
Looks great, thanks @TisonKun!

I'm going to merge this now and we can update docs after Java 11 tests are passing. Looks like we are still seeing failures on #577.

**Pull request merged**